packages/infra/lib/surescripts-stack.ts

```diff
@@ -0,0 +1,486 @@
+  const parseResponseLambdaTimeout = Duration.minutes(15);
+  const parseResponse: QueueAndLambdaSettings = {
+    name: "SurescriptsResponse",
+    entry: "surescripts-response",
+    lambda: {
+      memory: 1024,
+      timeout: parseResponseLambdaTimeout,
+    },
+    queue: {
+      alarmMaxAgeOfOldestMessage: Duration.hours(1),
+      maxMessageCountAlarmThreshold: 15_000,
+      maxReceiveCount: 3,
+      visibilityTimeout: Duration.seconds(parseResponseLambdaTimeout.toSeconds() * 2 + 1),
+      createRetryLambda: false,
+    },
+    eventSource: {
+      batchSize: 1,
+      reportBatchItemFailures: true,
+      maxConcurrency: 4,
+    },
+    waitTime: waitTimeRefreshBundle,
+  };
+
+  return {
+    connectSftp,
+    synchronizeSftp,
+    generateRequest,
+    parseVerification,
+    parseResponse,
+  };
+}
+
+interface SurescriptsNestedStackProps extends NestedStackProps {
+  config: EnvConfig;
+  vpc: ec2.IVpc;
+  alarmAction?: SnsAction;
+  lambdaLayers: LambdaLayers;
+  surescriptsReplicaBucket: s3.Bucket;
+  surescriptsBundleBucket: s3.Bucket;
+}
+
+export class SurescriptsNestedStack extends NestedStack {
+  readonly connectSftpLambda: Lambda;
+  readonly synchronizeSftpLambda: Lambda;
+  readonly generateRequestLambda: Lambda;
+  readonly generateRequestQueue: Queue;
+  readonly parseVerificationLambda: Lambda;
+  readonly parseVerificationQueue: Queue;
+  readonly parseResponseLambda: Lambda;
+  readonly parseResponseQueue: Queue;
+
+  readonly surescriptsBundleBucket: s3.Bucket;
+  readonly surescriptsReplicaBucket: s3.Bucket;
+
+  constructor(scope: Construct, id: string, props: SurescriptsNestedStackProps) {
+    super(scope, id, props);
+
+    this.terminationProtection = true;
+
+    const surescriptsBundleBucket = new s3.Bucket(this, "SurescriptsBundleBucket", {
+      bucketName: props.config.surescripts?.surescriptsBundleBucketName,
+      publicReadAccess: false,
+      encryption: s3.BucketEncryption.S3_MANAGED,
+      versioned: true,
+      cors: [
+        {
+          allowedOrigins: ["*"],
+          allowedMethods: [s3.HttpMethods.GET],
+        },
+      ],
+    });
+    this.surescriptsBundleBucket = surescriptsBundleBucket;
+
+    const surescriptsReplicaBucket = new s3.Bucket(this, "SurescriptsReplicaBucket", {
+      bucketName: props.config.surescripts?.surescriptsReplicaBucketName,
+      publicReadAccess: false,
+      encryption: s3.BucketEncryption.S3_MANAGED,
+      versioned: true,
+      cors: [
+        {
+          allowedOrigins: ["*"],
+          allowedMethods: [s3.HttpMethods.GET],
+        },
+      ],
+    });
+    this.surescriptsReplicaBucket = surescriptsReplicaBucket;
+
+    // Only scoped to read/write from the S3 bucket
+    const connectSftp = this.setupConnectSftp({
+      lambdaLayers: props.lambdaLayers,
+      vpc: props.vpc,
+      envType: props.config.environmentType,
+      sentryDsn: props.config.lambdasSentryDSN,
+      alarmAction: props.alarmAction,
```

Comment:
define these as "lambdaParams" or something and then reuse it, and spread it into each setup() function